src/Chirp.Infrastructure/Repositories/AuthorRepository.cs

```diff
@@ -122,5 +122,9 @@ public async Task UnfollowAuthor(Author author, Author authorToUnFollow)
 
             dbContext.AuthorAuthorRelations.Remove(authorAuthorRelation);
         }
+
```

Comment:
If author follows others, it seems that the author will not be deleted, because the authors still exists as a foreign key in the AuthorAuthorRelations.

This could be added to solve it (before deleting the author): 
```
var followers = await dbContext.AuthorAuthorRelations
 .Where(x => x.AuthorId == author.AuthorId || x.AuthorToFollowId == author.AuthorId)
 .ToListAsync();           

dbContext.AuthorAuthorRelations.RemoveRange(followers);
```